tests/unit_tests/collection/test_k_ohclv.py

```diff
@@ -0,0 +1,167 @@
+    # when
+    with pytest.raises(requests.exceptions.ConnectTimeout):
+        data = candles.fetch()
+    # then
+    assert candles.data == None
+
+def test_k_ohlcv_fetch_invalid_interval(requests_mock):
+    # given
+    mock_response = k_ohclv_success
+    endpoint = 'https://api.kraken.com/0/public/OHLC'
+    requests_mock.get(endpoint, json=mock_response)
+    candles = KrakenOHLCV('', 'XXBTZUSD', 1, 1560123060)
+    # when
+    error_msg_regx = re.compile('interval', re.IGNORECASE)
+    with pytest.raises(ValueError, match=error_msg_regx):
+        data = candles.fetch()
+    # then
+    assert candles.data == None
+
+# get * method tests
+
+def test_k_ohlcv_get_time(requests_mock):
+    # given
+    candles = KrakenOHLCV(Interval.MINUTE, 'XXBTZUSD', 1, 1560123060)
+    candles.data = k_ohclv_success_df
+    # when
+    data = candles.get_time()
+    # then
+    expected_data = k_ohclv_success_df['time']
+    assert_series_equal(data, expected_data)
+
+def test_k_ohlcv_get_open(requests_mock):
+    # given
+    candles = KrakenOHLCV(Interval.MINUTE, 'XXBTZUSD', 1, 1560123060)
+    candles.data = k_ohclv_success_df
+    # when
+    data = candles.get_open()
+    # then
+    expected_data = k_ohclv_success_df['open']
+    assert_series_equal(data, expected_data)
+
+def test_k_ohlcv_get_open(requests_mock):
+    # given
+    candles = KrakenOHLCV(Interval.MINUTE, 'XXBTZUSD', 1, 1560123060)
+    candles.data = k_ohclv_success_df
+    # when
+    data = candles.get_open()
+    # then
+    expected_data = k_ohclv_success_df['open']
+    assert_series_equal(data, expected_data)
+
+def test_k_ohlcv_get_high(requests_mock):
+    # given
+    candles = KrakenOHLCV(Interval.MINUTE, 'XXBTZUSD', 1, 1560123060)
+    candles.data = k_ohclv_success_df
+    # when
+    data = candles.get_high()
+    # then
+    expected_data = k_ohclv_success_df['high']
+    assert_series_equal(data, expected_data)
+
+def test_k_ohlcv_get_low(requests_mock):
+    # given
+    candles = KrakenOHLCV(Interval.MINUTE, 'XXBTZUSD', 1, 1560123060)
+    candles.data = k_ohclv_success_df
+    # when
+    data = candles.get_low()
+    # then
+    expected_data = k_ohclv_success_df['low']
+    assert_series_equal(data, expected_data)
+
+def test_k_ohlcv_get_close(requests_mock):
+    # given
+    candles = KrakenOHLCV(Interval.MINUTE, 'XXBTZUSD', 1, 1560123060)
+    candles.data = k_ohclv_success_df
+    # when
+    data = candles.get_close()
+    # then
+    expected_data = k_ohclv_success_df['close']
+    assert_series_equal(data, expected_data)
+
+def test_k_ohlcv_get_volume(requests_mock):
+    # given
+    candles = KrakenOHLCV(Interval.MINUTE, 'XXBTZUSD', 1, 1560123060)
+    candles.data = k_ohclv_success_df
+    # when
+    data = candles.get_volume()
+    # then
+    expected_data = k_ohclv_success_df['volume']
+    assert_series_equal(data, expected_data)
+
+# write method tests
+
+def test_k_ohlcv_write(requests_mock, tmp_dir):
```

Comment:
If you're writing unit tests for there `write()` methods you can use this `tmp_dir` fixture to write to a temporary directory.

Reply:
They should work anywhere in our `tests` directory